pkg/exasol-rest-api/authorizer.go

```diff
@@ -0,0 +1,38 @@
+package exasol_rest_api
+
+import (
+	"fmt"
+	error_reporting_go "github.com/exasol/error-reporting-go"
+	"net/http"
+)
+
+// Authorizer is responsible for the API users' authorization
+type Authorizer interface {
+	// Authorize a token
+	Authorize(request *http.Request) error
+}
+
+// TokenAuthorizer is a token-based implementation of the Authorizer
+type TokenAuthorizer struct {
+	AllowedTokens map[string]bool
+}
+
+func (auth *TokenAuthorizer) Authorize(request *http.Request) error {
+	tokens := request.Header["Authorization"]
+
+	authorized := false
+	for _, token := range tokens {
+		if auth.AllowedTokens[token] {
+			authorized = true
+			break
+		}
+	}
+
+	if !authorized {
+		return fmt.Errorf(error_reporting_go.ExaError("E-ERA-22").
```

Comment:
Wait 100ms here to slow down brute force attacks.

Reply:
The API deserves to have rate limits/throttling -> added to https://github.com/exasol/exasol-rest-api/issues/22